python/lsst/ip/diffim/dipoleFitTask.py

```diff
@@ -0,0 +1,921 @@
+        if (rel_weight > 0. and fitBgGradient and bgGradientOrder >= 0):
+            pbg = 0.
+            # Fit the gradient to the background (linear model)
+            bgParsPos = bgParsNeg = self.fitBackgroundGradient(source, self.posImage,
+                                                               order=bgGradientOrder)
+            # Generate the gradient and subtract it from the pre-subtraction image data
+            in_x = self._generateXYGrid(bbox)
+            pbg = self.genBgGradientModel(in_x, tuple(bgParsPos))
+            z[1, :] -= pbg
+            z[1, :] -= np.nanmedian(z[1, :])
+            posFlux = np.nansum(z[1, :])
+            gmod.set_param_hint('flux', value=posFlux*1.5, min=0.1)
+
+            if separateNegParams and self.negImage is not None:
+                bgParsNeg = self.fitBackgroundGradient(source, self.negImage, order=bgGradientOrder)
+                pbg = self.genBgGradientModel(in_x, tuple(bgParsNeg))
+            z[2, :] -= pbg
+            z[2, :] -= np.nanmedian(z[2, :])
+            if separateNegParams:
+                negFlux = np.nansum(z[2, :])
+                gmod.set_param_hint('fluxNeg', value=negFlux*1.5, min=0.1)
+
+            if False:  # we have subtracted the background from the images so dont fit anymore (faster!)
+                if bgGradientOrder >= 0:
+                    gmod.set_param_hint('b', value=bgParsPos[0])
+                    if separateNegParams:
+                        gmod.set_param_hint('bNeg', value=bgParsNeg[0])
+                if bgGradientOrder >= 1:
+                    gmod.set_param_hint('x1', value=bgParsPos[1])
+                    gmod.set_param_hint('y1', value=bgParsPos[2])
+                    if separateNegParams:
+                        gmod.set_param_hint('x1Neg', value=bgParsNeg[1])
+                        gmod.set_param_hint('y1Neg', value=bgParsNeg[2])
+                if bgGradientOrder >= 2:
+                    gmod.set_param_hint('xy', value=bgParsPos[3])
+                    gmod.set_param_hint('x2', value=bgParsPos[4])
+                    gmod.set_param_hint('y2', value=bgParsPos[5])
+                    if separateNegParams:
+                        gmod.set_param_hint('xyNeg', value=bgParsNeg[3])
+                        gmod.set_param_hint('x2Neg', value=bgParsNeg[4])
+                        gmod.set_param_hint('y2Neg', value=bgParsNeg[5])
+
+        y, x = np.mgrid[bbox.getBeginY():bbox.getEndY(), bbox.getBeginX():bbox.getEndX()]
+        in_x = np.array([x, y]).astype(np.float)
+        in_x[0, :] -= in_x[0, :].mean()  # center it!
+        in_x[1, :] -= in_x[1, :].mean()
+
+        # Instead of explicitly using a mask to ignore flagged pixels, just set the ignored pixels'
+        #  weights to 0 in the fit. Right now, this is not used. TBD: need to inspect mask planes to set
+        #  this mask.
+        mask = np.ones_like(z, dtype=bool)  # TBD: set mask values to False if the pixels are to be ignored
```

Comment:
You're correct, this is a placeholder for a future ticket. I've modified the comment and opened a ticket: DM-5857.